lighthouse-core/audits/timing-budget.js

```diff
@@ -0,0 +1,140 @@
+      'interactive': i18n.UIStrings.interactiveMetric,
+      'first-meaningful-paint': i18n.UIStrings.firstMeaningfulPaintMetric,
+      'max-potential-fid': i18n.UIStrings.maxPotentialFIDMetric,
+      'estimated-input-latency': i18n.UIStrings.estimatedInputLatencyMetric,
+      'total-blocking-time': i18n.UIStrings.totalBlockingTimeMetric,
+      'speed-index': i18n.UIStrings.speedIndexMetric,
+    };
+    return strMappings[timingMetric];
+  }
+
+  /**
+   * @param {LH.Budget.TimingMetric} timingMetric
+   * @param {LH.Artifacts.TimingSummary} summary
+   * @return {number|undefined}
+   */
+  static getMeasurement(timingMetric, summary) {
+    /** @type {Record<LH.Budget.TimingMetric,number|undefined>} */
+    const measurements = {
+      'first-contentful-paint': summary.firstContentfulPaint,
+      'first-cpu-idle': summary.firstCPUIdle,
+      'interactive': summary.interactive,
+      'first-meaningful-paint': summary.firstMeaningfulPaint,
+      'max-potential-fid': summary.maxPotentialFID,
+      'estimated-input-latency': summary.estimatedInputLatency,
+      'total-blocking-time': summary.totalBlockingTime,
+      'speed-index': summary.speedIndex,
+    };
+    return measurements[timingMetric];
+  }
+
+  /**
+     * @param {LH.Budget} budget
+     * @param {LH.Artifacts.TimingSummary} summary
+     * @return {Array<BudgetItem>}
+     */
+  static tableItems(budget, summary) {
+    if (!budget.timings) {
+      return [];
+    }
+    return budget.timings.map((timingBudget) => {
+      const metric = timingBudget.metric;
+      const label = str_(this.getRowLabel(timingBudget.metric));
+      const measurement = this.getMeasurement(timingBudget.metric, summary);
+      const overBudget = measurement && (measurement > timingBudget.budget)
+        ? (measurement - timingBudget.budget) : undefined;
+      return {
+        metric,
+        label,
+        measurement,
+        overBudget,
+      };
+    }).sort((a, b) => {
+      return (b.overBudget || 0) - (a.overBudget || 0);
+    });
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const mainResource = await MainResource.request({URL: artifacts.URL, devtoolsLog}, context);
+    const summary = (await TimingSummary.request({trace, devtoolsLog}, context)).metrics;
+    const budget = Budget.getMatchingBudget(context.settings.budgets, mainResource.url);
+
+    if (!budget) {
+      return {
+        score: 0,
+        notApplicable: true,
+      };
+    }
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headers = [
+      {key: 'label', itemType: 'text', text: str_(UIStrings.columnTimingMetric)},
+      {key: 'measurement', itemType: 'ms', text: str_(i18n.UIStrings.columnDuration)},
```

Comment:
I wonder if we should use `Measurement`/`Value`/`Timing` or something instead. while the units are all labeled with time, it feels weird in my mind to call speed index a duration.